tool/gocross/autoflags.go

```diff
@@ -0,0 +1,183 @@
+		buildFlags  = []string{"-trimpath"}
+		cgoCflags   = []string{"-O3", "-std=gnu11"}
+		cgoLdflags  []string
+		ldflags     []string
+		tags        = []string{"tailscale_go"}
+		cgo         = false
+		failReflect = false
+	)
+	if len(argv) > 1 {
+		subcommand = argv[1]
+	}
+
+	switch subcommand {
+	case "build", "env", "install", "run", "test", "list":
+	default:
+		return argv, env, nil
+	}
+
+	vi := getVersion()
+	ldflags = []string{
+		"-X", "tailscale.com/version.longStamp=" + vi.Long,
+		"-X", "tailscale.com/version.shortStamp=" + vi.Short,
+		"-X", "tailscale.com/version.gitCommitStamp=" + vi.GitHash,
+		"-X", "tailscale.com/version.extraGitCommitStamp=" + vi.OtherHash,
+	}
+
+	switch targetOS {
+	case "linux":
+		// Getting Go to build a static binary with cgo enabled is a
+		// minor ordeal. The incantations you apparently need are
+		// documented at: https://github.com/golang/go/issues/26492
+		tags = append(tags, "osusergo", "netgo")
+		cgo = targetOS == nativeGOOS && targetArch == nativeGOARCH
+		// When in a Nix environment, the gcc package is built with only dynamic
+		// versions of glibc. You can get a static version of glibc via
+		// pkgs.glibc.static, but then you are reliant on Nix's gcc wrapper
+		// magic to inject that as a -L path to linker invocations.
+		//
+		// We can't rely on that magic linker flag injection, because that
+		// injection breaks redo's go machinery for dynamic go+cgo linking due
+		// to flag ordering issues that we can't easily fix (since the nix
+		// machinery controls the flag ordering, not us).
+		//
+		// So, instead, we unset NIX_LDFLAGS in our nix shell, which disables
+		// the magic linker flag passing; and we have shell.nix drop the path to
+		// the static glibc files in GOCROSS_GLIBC_DIR. Finally, we reinject it
+		// into the build process here, so that the linker can find static glibc
+		// and complete a static-with-cgo linkage.
+		extldflags := []string{"-static"}
+		if glibcDir := env.Get("GOCROSS_GLIBC_DIR", ""); glibcDir != "" {
+			extldflags = append(extldflags, "-L", glibcDir)
+		}
+		// -extldflags, when it contains multiple external linker flags, must be
+		// quoted in its entirety as a member of -ldflags. Source:
+		// https://github.com/golang/go/issues/6234
+		ldflags = append(ldflags, fmt.Sprintf("'-extldflags=%s'", strings.Join(extldflags, " ")))
+	case "windowsgui":
+		// Fake GOOS that translates to "windows, but building GUI .exes not console .exes"
+		targetOS = "windows"
+		ldflags = append(ldflags, "-H", "windowsgui", "-s")
+	case "windows":
+		ldflags = append(ldflags, "-H", "windows", "-s")
+	case "ios":
+		failReflect = true
+		fallthrough
+	case "darwin":
+		cgo = nativeGOOS == "darwin"
+		tags = append(tags, "omitidna", "omitpemdecrypt")
+		if env.IsSet("XCODE_VERSION_ACTUAL") {
+			var xcodeFlags []string
+			// Minimum OS version being targeted, results in
+			// e.g. -mmacosx-version-min=11.3
+			minOSKey := env.Get("DEPLOYMENT_TARGET_CLANG_FLAG_NAME", "")
```

Comment:
Having the default be "" means xcode will be passed an argument of `-=`, which seems like it cannot work. Should we hard-code a minimum iOS and macOS version here?

Reply:
AFAICT, Xcode can't not set these envvars. And if it does stop setting them, the compile here will fail because `-mmacosx-version-min=""` isn't valid. I think I'd rather thave that failure mode, rather than silently paper over Xcode behavioral change and not figure out mysterious problems for a long time.